modules/network/pre-existing-vpc/main.tf

```diff
@@ -18,10 +18,24 @@
 data "google_compute_network" "vpc" {
   name    = var.network_name
   project = var.project_id
+
+  lifecycle {
+    postcondition {
+      condition     = self.name != null
+      error_message = "The network provided could not be found."
+    }
+  }
 }
 
 data "google_compute_subnetwork" "primary_subnetwork" {
   name    = var.subnetwork_name != null ? var.subnetwork_name : var.network_name
   region  = var.region
   project = var.project_id
+
+  lifecycle {
+    postcondition {
+      condition     = self.name != null
+      error_message = "The network provided could not be found."
```

Comment:
error message should state "subnetwork". Also same as above, could we print the variables that were used in the search.

Reply:
Done

Reply:
Can we print all the information used for lookup. for example: `The subnetwork: ${local.subnetwork_name} could not be found in project: ${var.project_id} and region: ${var.region}.`